### PR TITLE
DAOS-16487 control: Require hostname for nvme set-faulty & replace

### PR DESCRIPTION
**Pull request by tanabarr**
Makes --host option mandatory for dmg storage set nvme-faulty, dmg
storage replace nvme and dmg telemetry metrics. These commands should
only be run on a single host which should be explicitly specified.

Also remove --no-reint flag from replace nvme command.

Features: control
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/dmg_utils_base.py [hunk 1]
```diff
@@ -685,7 +685,7 @@ def __init__(self):
                     super().__init__("/run/dmg/storage/replace/nvme/*", "nvme")
                     self.old_uuid = FormattedParameter("--old-uuid {}", None)
                     self.new_uuid = FormattedParameter("--new-uuid {}", None)
-                    self.no_reint = FormattedParameter("--no-reint", False)
+                    self.host = FormattedParameter("--host {}", None)
```
Comment (kjacque) [inline src/tests/ftest/util/dmg_utils_base.py:688]: Need to update dmg args for the telemetry query as well? I'd suggest running tests for the telemetry feature to ensure nothing was broken.

**Review by kjacque — COMMENTED** [review]
Overall looks good. Ftest dmg infrastructure updates may be needed for the telemetry commands now that the single host command has changed.

File: src/tests/ftest/util/dmg_utils.py [hunk 2]
```diff
@@ -252,16 +252,18 @@ def storage_format(self, force=False, timeout=30, verbose=False):
         self.timeout = saved_timeout
         return self.result
 
-    def storage_set_faulty(self, uuid, force=True):
+    def storage_set_faulty(self, host, uuid, force=True):
```
Comment (daltonbohning) [inline src/tests/ftest/util/dmg_utils.py:255]: Also need to update this usage
https://github.com/daos-stack/daos/blob/f9b3f46a3d0e2b6c7555d9e4897d0eba398feca7/src/tests/ftest/control/dmg_storage_query.py#L255

File: src/tests/ftest/util/dmg_utils.py [hunk 3]
```diff
@@ -338,13 +340,13 @@ def storage_led_check(self, ids=None):
         return self._get_json_result(
             ("storage", "led", "check"), ids=ids)
 
-    def storage_replace_nvme(self, old_uuid, new_uuid, no_reint=False):
+    def storage_replace_nvme(self, host, old_uuid, new_uuid):
```
Comment (daltonbohning) [inline src/tests/ftest/util/dmg_utils.py:343]: Also need to update this usage
https://github.com/daos-stack/daos/blob/f9b3f46a3d0e2b6c7555d9e4897d0eba398feca7/src/tests/ftest/vmd/led.py#L133

**Review by knard-intel — COMMENTED** [review]
LGTM.
+1 with kris for adding the telemetry feature.

File: src/tests/ftest/util/dmg_utils.py (hunk 2 above)
Comment (tanabarr) [reply src/tests/ftest/util/dmg_utils.py:255]: good catch, thanks. fixed

File: src/tests/ftest/util/dmg_utils.py (hunk 3 above)
Comment (tanabarr) [reply src/tests/ftest/util/dmg_utils.py:343]: good catch, thanks. fixed

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Review by knard-intel — DISMISSED** [review]

**Comment by daltonbohning** [comment]
deployment/disk_failure.py and vmd/fault_reintegration.py failures in
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-15074/5/tests
are manual tests and not expected to pass in CI

File: src/tests/ftest/util/dmg_utils_base.py (hunk 1 above)
Comment (mjmac) [reply src/tests/ftest/util/dmg_utils_base.py:688]: Was this addressed?

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

File: src/tests/ftest/util/dmg_utils_base.py (hunk 1 above)
Comment (tanabarr) [reply src/tests/ftest/util/dmg_utils_base.py:688]: yes and telemetry feature is being run in CI

File: src/tests/ftest/util/nvme_utils.py [hunk 4]
```diff
@@ -44,7 +44,7 @@ def set_device_faulty(test, dmg, server, uuid, pool=None, has_sys_xs=False, **kw
     Args:
         test (Test): avocado test class
         dmg (DmgCommand): a DmgCommand class instance
-        server (NodeSet): host on which to issue the dmg storage set nvme-faulty
+        server (str): host on which to issue the dmg storage set nvme-faulty
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/tests/ftest/util/nvme_utils.py]: fixed

**Review by mjmac — COMMENTED** [review]
Changes look good to me, but I'll defer to those with more context into requirements for approvals.

**Review by kjacque — APPROVED** [review]
Looks OK to me as long as telemetry tests pass.

**Review by knard-intel — APPROVED** [review]

**Pull request merged**